exercises/pythagorean-triplet/pythagorean-triplet.test.ts

```diff
@@ -0,0 +1,37 @@
+import Triplet from './pythagorean-triplet'
+
+describe('Triplet', () => {
+    it('calculates the sum', () => {
+        expect(new Triplet(3, 4, 5).sum()).toBe(12)
+    })
+
+    xit('calculates the product', () => {
+        expect(new Triplet(3, 4, 5).product()).toBe(60)
+    })
+
+    xit('can recognize a pythagorean triplet', () => {
+        expect(new Triplet(3, 4, 5).isPythagorean()).toBe(true)
+    })
+
+    xit('can recognize a non pythagorean triplet', () => {
+        expect(new Triplet(5, 6, 7).isPythagorean()).toBe(false)
+    })
+
+    xit('can make triplets up to 10', () => {
+        const triplets = Triplet.where({ maxFactor: 10 })
```

Comment:
I'd rather have `where( maxFactor: number, minFactor?: number = null, sum?: number = null)` or similar than having to pass in an object. 

Are you trying to mimic named params? https://stackoverflow.com/questions/42108807/using-named-parameters-javascriptbased-on-typescript

Reply:
I've swapped out the object for optional parameters as suggested. The object was how it was implemented in the ecmascript exercise, which I agree doesn't make as much sense in TS.